File: room_demo/src/main/java/com/xiongtao/room_demo/MainActivity2.java
package com.xiongtao.room_demo;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProvider;
import androidx.lifecycle.ViewModelProviders;

import android.os.Bundle;
import android.widget.ListView;

import com.xiongtao.room_demo.entity.Student;

import java.util.List;

public class MainActivity2 extends AppCompatActivity {
    ListView listView;
    StudentViewModel studentViewModel;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        listView = findViewById(R.id.listView);

        studentViewModel = ViewModelProviders.of(this).get(StudentViewModel.class);
        studentViewModel.getAllLiveDataStudent().observe(this, new Observer<List<Student>>() {
            @Override
            public void onChanged(List<Student> students) {
                // 更新UI
                listView.setAdapter(new GoodsAdapter(MainActivity2.this, students));
            }
        });


        // 模拟 仓库
        new Thread()  {
            @Override
            public void run() {
                super.run();

                try {
                    Thread.sleep(6000);
                } catch (InterruptedException e) {
                    e.printStackTrace();
                }

                // 默认给数据库ROOM增加数据
                for (int i = 0; i < 50; i++) {
                    studentViewModel.insert(new Student("Derry", "123", 1));
                }
            }
        }.start();


        // 模拟仓库 数据库 数据被修改了，一旦数据库被修改，那么数据会驱动UI的发生改变
        new Thread() {
            @Override
            public void run() {
                for (int i = 0; i < 50; i++) {
                    try {
                        Thread.sleep(1000);
                    } catch (Exception e) {
                        e.printStackTrace();
                    }
                    studentViewModel.update(new Student(6,"Derry" + i, "123", 1));
                }
            }
        }.start();

    }
}

File: livedata_demo/src/main/java/com/xiongtao/livedata_demo/simple4/MainActivity2.kt
package com.xiongtao.livedata_demo.simple4

import android.os.Bundle
import android.util.Log
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.xiongtao.livedata_demo.R
import kotlin.concurrent.thread

/**
 * @Description:
 * @Author: xiongtao
 * @Date: 2023-03-07 15:49
 */
class MainActivity2: AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main2)

        OKLiveDataBusKT.with("data1",String::class.java).observe(this,{
            Toast.makeText(this, "Kotlin版本的 观察者:${it}", Toast.LENGTH_SHORT).show()
            Log.d("derry", "Kotlin版本的 观察者:${it}")
        })

        // Java版本订阅观察者(Java是模拟就剔除黏性的，你自己增加开关)
        OKLiveDataBusJava.getInstance().with("data2", String::class.java).observe(this, {
            Toast.makeText(this, "Java版本的 观察者:${it}", Toast.LENGTH_SHORT).show()
            Log.d("derry", "Java版本的 观察者:${it}")
        })

        thread {
            Thread.sleep(6000)
            OKLiveDataBusKT.with("data1", String::class.java).postValue("666")
        }

        thread {
            Thread.sleep(12000)
            OKLiveDataBusJava.getInstance().with("data2", String::class.java).postValue("121212")
        }
    }
}